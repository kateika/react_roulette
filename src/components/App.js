import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import '../styles/normalize.css';
import  '../styles/common.css';
import ListResultContainer from '../containers/ListResultContainer';
import FilmDescriptionContainer from '../containers/FilmDescriptionContainer';
import NotFound from '../components/NotFound';

export default function App() {
  return (
    <Router>
      <Switch>
        <Route exact path="/" component={ListResultContainer} />
        <Route exact path="/search" component={ListResultContainer} />
        <Route path="/search/:searchQuery" component={ListResultContainer} />
        <Route path="/:type/:filmId/" component={FilmDescriptionContainer} />
        <Route path="*" component={NotFound} />
      </Switch>
    </Router>
  )
}
