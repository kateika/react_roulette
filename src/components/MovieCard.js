import React from 'react';
import { Link } from 'react-router-dom';
import * as css from '../styles/movie-card.css';

export class MovieCard extends React.Component {
  render() {
    const { poster, title, release_date, id } = this.props;
    return (
      <Link to={`/film/${id}`} className={css.card}>
        <div className={css.content}>
          <img src={poster}/>
          <div className="clearfix">
            <h4 className={css.title}>{title}</h4>
            {release_date ? <span className={css.year}>{release_date}</span> : null}
          </div>
        </div>
      </Link>
    )
  }
};
