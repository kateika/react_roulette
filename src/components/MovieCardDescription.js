import React from 'react';
import * as css from '../styles/movie-card-description.css';

export class MovieCardDescription extends React.Component {
  render() {
    const { poster, title, vote_average, release_date, runtime, overview, budget, seasons, last_air_date } = this.props.currentMovie;
    const rating = <div className={css.ratingCircle}>
                     <span className={css.rating}>{vote_average}</span>
                   </div>;
    return (
      <article className={css.container + " clearfix"}>
        {poster ? <img src={`https://image.tmdb.org/t/p/w185_and_h278_bestv2${poster}`} className={css.poster}/> : null}
        <div className={css.description}>
          <h4 className={css.title}>{title}
            { vote_average ? rating : null }
          </h4>

          <div className={css.boldCaption}>
            { release_date ? <span className={css.year}>{release_date}</span> : null }
            { runtime ? <span>{runtime} min</span> : null }
          </div>
          <p>{overview}</p>
          { budget ? <span className={css.caption}>Budget is {budget}$</span> : null }
          { seasons ? <span className={css.caption}>Seasons: {seasons}</span> : null }
          { last_air_date ? <span className={css.caption}>Last air date: {last_air_date}</span> : null }
        </div>
      </article>
    )
  }
};

