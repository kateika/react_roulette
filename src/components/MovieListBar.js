import React from 'react';
import * as css from '../styles/movie-list.css';

export class MovieListBar extends React.Component {
  render() {
    return (
      <div className={css.resultsBar}>
        <div className={css.resultsContainer + " container"}>
          {this.props.length == 1 ? <span>1 result was found</span> : <span>{this.props.length} results was found</span>}
          <div className={css.sorting}>
            <span>Sort by:</span>
            <a href="#">release date</a>
            <a href="#" className={css.linkActive}>rating</a>
          </div>
        </div>
      </div>
    )
  }
};
