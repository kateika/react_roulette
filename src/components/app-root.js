import React from "react";
import { renderRoutes } from "react-router-config";

const AppRoot = (props) => {
  console.log(props.route);
  return (
    <div>
      {renderRoutes(props.route.routes)}
    </div>
  );
};

export default AppRoot;
