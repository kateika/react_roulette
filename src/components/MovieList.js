import React from 'react';
import { MovieCard } from './MovieCard';
import * as css from '../styles/start-page.css';
import * as cssList from '../styles/movie-list.css';

export class MovieList extends React.Component {
  render() {
    function extractYear(dateString) {
      if(dateString) {
        let year = dateString.slice(0,4);
        return year;
      }
    }

    const movies = this.props.movies.map((movie, index) => {
      return <MovieCard
        title={movie.title}
        release_date={extractYear(movie.release_date)}
        poster={`https://image.tmdb.org/t/p/w185_and_h278_bestv2${movie.poster}`}
        id={movie.id}
        type={movie.type}
        key={index}
      />
    });

    const noMovies = <div className={css.container}>No films found</div>;

    return (
      <div>
        <div className={cssList.resultsBar}>
          <div className={cssList.resultsContainer + " container"}>
            {movies.length ? this.props.children : null}
          </div>
        </div>
        <div className="flex container relative">
          { movies.length ? movies : noMovies}
        </div>
      </div>
    )
  }
};
