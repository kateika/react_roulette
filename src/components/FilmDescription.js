import React from 'react';
import { Link } from 'react-router-dom';
import { MovieList } from './MovieList';
import { MovieCardDescription } from './MovieCardDescription';


export class FilmDescription extends React.Component {
  //static fetchData(store) {
  //  return store.dispatch(props.loadMovieInfo(props.match.params.filmId, props.match.params.type));
  //}

  componentWillMount() {
    this.props.loadMovieInfo(this.props.match.params.filmId, this.props.match.params.type);
  }

  componentWillReceiveProps(nextProps) {
    if(nextProps.match.params.filmId != this.props.match.params.filmId) {
      this.props.loadMovieInfo(nextProps.match.params.filmId, this.props.match.params.type);
    }
  }

  render() {
    let movieListBar = `Films by ${this.props.currentMovie.director}`;
    return (
      <div>
        <div className="page-wrapper">
          <header className="header">
            <div className="container">
              <div className="top-header clearfix">
                <h1 className="logo"><Link to="/">netflixroulette</Link></h1>
                <Link to="/search" className="btn secondary-btn">Search</Link>
              </div>
              <MovieCardDescription currentMovie={this.props.currentMovie} />
            </div>
          </header>
          <main>
            <MovieList movies={this.props.relatedMovies}>
              {this.props.currentMovie.director ? movieListBar : null}
            </MovieList>
          </main>
        </div>
        <footer className="footer">
          <div className="container">
            <h1 className="logo"><Link to="/">netflixroulette</Link></h1>
            <span className="api-logo">This product uses the TMDb API but is not endorsed or certified by TMDb.</span>
          </div>
        </footer>
      </div>
    )
  }
};
