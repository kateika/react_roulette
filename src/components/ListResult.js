import React from 'react';
import { Link } from 'react-router-dom';
import { SearchForm } from '../containers/SearchForm';
import { Sorting } from '../containers/Sorting';
import { MovieList } from './MovieList';
import * as cssList from '../styles/movie-list.css';

export class ListResult extends React.Component {
  render() {
    const resultQuentity = this.props.movies.length;
    const results = resultQuentity == 1 ? "result" : "results";
    const movieListBar =  <span>{resultQuentity} {results} was found</span>;

    return (
      <div>
        <div className="page-wrapper">
          <header className="header">
            <div className="container">
              <div className="top-header">
                <h1 className="logo"><Link to="/">netflixroulette</Link></h1>
              </div>
              <SearchForm />
            </div>
          </header>
          <main>
            <MovieList movies={this.props.movies}>
              { movieListBar }
              <Sorting />
            </MovieList>
          </main>
        </div>
        <footer className="footer">
          <div className="container">
            <h1 className="logo"><Link to="/">netflixroulette</Link></h1>
          </div>
        </footer>
      </div>
    )
  }
};
