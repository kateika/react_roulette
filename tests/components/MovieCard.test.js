import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { mount, configure } from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import { MovieCard } from "../../src/components/MovieCard";
import Adapter from 'enzyme-adapter-react-15';

configure({ adapter: new Adapter() });

const title= "test title";
const release_date= "1976-06-17";
const poster= "https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg";
const id= 111;
const type= "movie";

describe("Movie card component", () => {
  test("renders valid layout when we receive all props", () => {
    const tree = ReactTestRenderer.create(
      <MemoryRouter>
        <MovieCard title={title} release_date={release_date} poster={poster} type={type} id={id}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders valid layout without release_date", () => {
    const tree = ReactTestRenderer.create(
      <MemoryRouter>
        <MovieCard title={title} poster={poster} type={type} id={id}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("constructs link for router", () => {
    const wrapper = mount(
      <MemoryRouter>
        <MovieCard type={type} id={id}/>
      </MemoryRouter>
    );

    expect(wrapper.find('[href="/movie/111"]').length).toBe(1);
  });
});
