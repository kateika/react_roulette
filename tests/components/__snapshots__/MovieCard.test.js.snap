// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card component renders valid layout when we receive all props 1`] = `
<a
  className="card"
  href="/movie/111"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <img
      src="https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg"
    />
    <div
      className="clearfix"
    >
      <h4
        className="title"
      >
        test title
      </h4>
      <span
        className="year"
      >
        1976-06-17
      </span>
    </div>
  </div>
</a>
`;

exports[`Movie card component renders valid layout without release_date 1`] = `
<a
  className="card"
  href="/movie/111"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <img
      src="https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg"
    />
    <div
      className="clearfix"
    >
      <h4
        className="title"
      >
        test title
      </h4>
    </div>
  </div>
</a>
`;
